### PR TITLE
Diatonic and non-diatonic scales, Key module

### PR DESCRIPTION
**Pull request by duncanmalashock**

### CODE REVIEW

File: src/MusicTheory/Key.elm [hunk 1]
```diff
@@ -0,0 +1,50 @@
+module Key exposing
```
Comment (battermann) [inline src/MusicTheory/Key.elm]: This module looks quite nice.

We could provide functions that return the corresponding/relative minor to a major key and vice versa.

Comment (battermann) [inline src/MusicTheory/Key.elm]: The module name should be `MusicTheory.Key`.

File: src/MusicTheory/Key.elm [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+import MusicTheory.PitchClass as PitchClass exposing (PitchClass)
+import MusicTheory.Scale as Scale
+
+
+type Key
+    = Key Tonality DiatonicScaleDegrees
+
+
+type Tonality
```
Comment (battermann) [inline src/MusicTheory/Key.elm:17]: I don't think this is the right name. Tonality involves a tonic but this just specifies if it's minor or major.

We could call this `KeyClass` or `KeyQuality`.

File: src/MusicTheory/Key.elm [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+minor : PitchClass -> Key
+minor root =
+    Key Minor
+        (Scale.diatonicScale root Scale.minor
+            |> Scale.diatonicScaleDegrees
+        )
+
+
+scaleDegrees : Key -> DiatonicScaleDegrees
```
Comment (battermann) [inline src/MusicTheory/Key.elm:42]: I think it's good that the codomain restricted to `DiatonicScaleDegrees`.

But ATM there is no way to make either a list or a `ScaleClass` out of this (at least I didn't find a way). There should be some way to do this.

**Review by battermann — CHANGES_REQUESTED** [review]
So far I only review the Key module.

**Comment by duncanmalashock** [comment]
After discussing with @battermann I agree that this implementation is too complicated. Closing.

**Pull request closed without merging**